airbyte-integrations/connectors/destination-iceberg-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/iceberg/v2/IcebergV2DataDumper.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.iceberg.v2
+
+import io.airbyte.cdk.command.ConfigurationSpecification
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.data.*
+import io.airbyte.cdk.load.data.parquet.ParquetMapperPipelineFactory
+import io.airbyte.cdk.load.message.DestinationRecord
+import io.airbyte.cdk.load.test.util.DestinationDataDumper
+import io.airbyte.cdk.load.test.util.OutputRecord
+import io.airbyte.integrations.destination.iceberg.v2.io.IcebergUtil
+import io.airbyte.protocol.models.v0.AirbyteRecordMessageMetaChange
+import java.math.BigDecimal
+import java.time.Instant
+import java.util.LinkedHashMap
+import java.util.UUID
+import org.apache.hadoop.conf.Configuration
+import org.apache.iceberg.catalog.TableIdentifier
+import org.apache.iceberg.data.IcebergGenerics
+import org.apache.iceberg.data.Record
+import org.apache.iceberg.nessie.NessieCatalog
+
+object IcebergV2DataDumper : DestinationDataDumper {
+
+    private fun convert(value: Any?, type: AirbyteType): AirbyteValue {
```

Comment:
(nonblocking) let's merge this as-is, but I want to try a different structure at some point:
* rely solely on the type of `value` to generate an AirbyteValue (possibly with different types than expected)
* implement an `ExpectedRecordMapper` to convert the expected record's types to match the dumped values

(not entirely sure which way is better - like I said, this code is still pretty young)